### ISSUE TITLE
Wrong aggregation result?

### ISSUE THREAD

**Issue by AlexMAS** [ClickHouse/ClickHouse#2509]
I have an **empty** table:

```sql
create table history
(
    sensorId String,
    date Date default toDate(time),
    time DateTime,
    value Float64
)
ENGINE = MergeTree(date, (sensorId, date, time), 8192)
```

When I execute the query:

```sql
select min(time) as t from history
```

I expect `NULL` but receive `0`:

```
┌───────────────────t─┐
│ 0000-00-00 00:00:00 │
└─────────────────────┘

1 rows in set. Elapsed: 0.004 sec.
```

This behaviour started recently after updating to the latest version - `1.1.54385`.

Is it correct behaviour?

**Comment by alexey-milovidov** [comment]
This is intended behaviour.

In previous version you get empty result for this query.
This was changed in version 1.1.54356:
https://github.com/yandex/ClickHouse/blob/master/CHANGELOG.md#new-features-3
You can turn back previous behaviour with setting "empty_result_for_aggregation_by_empty_set".

But current behaviour is still differs from what you would get in another DBMS,
because you get '0000-00-00 00:00:00' instead of NULL.

We return non-NULL value to avoid using Nullable types in typical cases.
Probably we should add an option to make every aggregate function returning Nullable type.

**Comment by AlexMAS** [comment]
Thank you for your reply. Yes, I assumed that was intentional changes. I've added processing for this case to my code so it isn't critical for me. I think we can close this issue.

**Comment by alexey-milovidov** [comment]
Now you can add -OrNull suffix to any aggregate function. For example, `minOrNull`.
We still don't have an option to make it default.

**Comment by alexey-milovidov** [comment]
Closed in favor of #10273.
